lib/logstash/outputs/syslog.rb

```diff
@@ -94,8 +94,12 @@ class LogStash::Outputs::Syslog < LogStash::Outputs::Base
   # syslog message format: you can choose between rfc3164 or rfc5424
   config :rfc, :validate => ["rfc3164", "rfc5424"], :default => "rfc3164"
 
+  # use codec to format message (instead of setting message)
+  config :usecodec, :validate => :boolean, :default => false
```

Comment:
I think this would be possible. The implication would be, that it wouldn't be possible to use the codec plain together with the syslog output. In my opinion this is not that bad because the functionality of the config option `message` of the syslog output is comparable to the plain codec.

@purbon what do you think?